indico/modules/events/registration/fields/simple.py

```diff
@@ -311,21 +310,23 @@ def process_formdata(self, valuelist):
 
 class FileField(RegistrationFormFieldBase):
     name = 'file'
-    wtf_field_class = _DeletableFileField
+    # Toggle this to make the angular form work
+    # wtf_field_class = _DeletableFileField
+    wtf_field_class = wtforms.StringField
 
     def process_form_data(self, registration, value, old_data=None, billable_items_locked=False):
         data = {'field_data': self.form_item.current_data}
         if not value:
             return data
-        file_ = value['uploaded_file']
-        if file_:
+
+        if file := File.query.filter(File.uuid == value).first():
             # we have a file -> always save it
             data['file'] = {
-                'data': file_.stream,
-                'name': secure_client_filename(file_.filename),
-                'content_type': mimetypes.guess_type(file_.filename)[0] or file_.mimetype or 'application/octet-stream'
+                'data': file.open(),
+                'name': file.filename,
+                'content_type': file.content_type
             }
-        elif not value['keep_existing']:
+        else:
```

Comment:
I don't think that's a good idea - you get in here if the uuid is invalid... but that should be an error; only if there's no UUID whatsoever (null / empty string atm) it should be considered "no file".

Reply:
How should the uuid errors be handled in `process_form_data`? raising `ValidationError` directly or defining a validator for the field?

Reply:
Since we'll eventually move to wtforms you could even just raise an exception and add a TODO comment to do it properly later